File: Practice/AOJ/DSL/DSL3_A/dsl3_a.php
<?php
  $ns = array_map("intVal", explode(" ", trim(fgets(STDIN))));
  $n = $ns[0];
  $s = $ns[1];
  $arr = array_map("intVal", explode(" ", trim(fgets(STDIN))));
  $INF = pow(2, 31) - 1;
  $ans = $INF;

  $tmp = 0;
  $r = 0;
  for($l=0; $l<$n; $l++){
    while($r < $n && $tmp < $s){
      $tmp += $arr[$r];
      $r += 1;
    }
    if($tmp >= $s){
      $ans = min($ans, $r - $l);
    }
    if($r == $l){
      $r += 1;
    }else{
      $tmp -= $arr[$l];
    }
  }

  echo $ans != $INF ? $ans . PHP_EOL : 0 . PHP_EOL;
?>

File: Practice/AOJ/DSL/DSL3_A/dsl3_a.py
# 尺取を使う その1
n, s = map(int, input().split())
arr = list(map(int, input().split()))
INF = 2 ** 31 - 1
ans = INF
tmp = 0
r = 0
for l in range(n):
    while r < n and tmp < s:
        tmp += arr[r]
        r += 1

    if tmp >= s:
        ans = min(ans, r - l)
    if r == l:
        r += 1
    else:
        tmp -= arr[l]

print(0 if ans == INF else ans)
